server/internal/usecase/operator.go

```diff
@@ -16,6 +20,60 @@ type Operator struct {
 	OwningProjects     project.IDList
 }
 
+type UserOperator struct {
+	Operator
+}
+
+type IntegrationOperator struct {
+	Operator
+}
+
+func (o *Operator) UserOperator() (*UserOperator, error) {
+	if o == nil || o.user == nil {
+		return nil, errors.New("invalid operator")
+	}
+	return &UserOperator{*o}, nil
+}
+
+func NewUserOperator(u user.ID, rw, ww, ow user.WorkspaceIDList, rp, wp, op project.IDList) *Operator {
+	return &Operator{
+		user:               &u,
+		ReadableWorkspaces: rw,
+		WritableWorkspaces: ww,
+		OwningWorkspaces:   ow,
+		ReadableProjects:   rp,
+		WritableProjects:   wp,
+		OwningProjects:     op,
+	}
+}
+
+func NewIntegrationOperator(i integration.ID, rw, ww, ow user.WorkspaceIDList, rp, wp, op project.IDList) *Operator {
+	return &Operator{
+		integration:        &i,
+		ReadableWorkspaces: rw,
+		WritableWorkspaces: ww,
+		OwningWorkspaces:   ow,
+		ReadableProjects:   rp,
+		WritableProjects:   wp,
+		OwningProjects:     op,
+	}
+}
+
+func (uo *UserOperator) User() user.ID {
+	return *uo.user
+}
+
+func (o *Operator) IntegrationOperator() (*IntegrationOperator, error) {
+	if o == nil || o.integration == nil {
+		return nil, errors.New("invalid operator")
+	}
+	return &IntegrationOperator{*o}, nil
+}
+
+func (io *IntegrationOperator) Integration() integration.ID {
+	return *io.integration
+}
+
```

Comment:
In light of my comment, I do not believe it is necessary to strictly separate operators by user and integration. So UserOperator and IntegrationOperator are not needed, I think.